### DISCUSSION TITLE
ITMS-90535: Unexpected CFBundleExecutable Key AppStore rejection

### DISCUSSION THREAD

**Discussion by antijingoist** [ionic-team/capacitor#6273] [Q&A] [answered]
Trying to submit an app for test flight, and it gets rejected. Every submission (troubleshoot, try past suggestion, resubmit or upgrade, resubmit) gets me the same message for each capacitor module: 

"ITMS-90535: Unexpected CFBundleExecutable Key: '{{path to plugin}}/Info.plist' does not contain a bundle executable. If this bundle intentionally does not contain an executable, consider removing the CFBundleExecutable key from its Info.plist and using a CFBundlePackageType of BNDL. If this bundle is part of a third-party framework, consider contacting the developer of the framework for an update to address this issue."

Removing the CFBundleExecutable causes the app to not launch.

Perhaps this is taken into account in the docs but having spent a week reading them I found nothing. (I've been working on it for some time in case it was on my end only. It still could be, but starting from scratch, updates, etc yields the same results).

**Comment by antijingoist** [comment, marked as answer]
*Attention people of the future* 

So after much discussion and help received in the Discord, it was a combination of things:

Needed to have CFBundleExecutable key removed from the info.plists in the public/node_modules folder, with CFBundlePackageType set to FMWK.

That change was messing up everything or not even happening because of Hard drive corruption. After that was fixed, I was able to remove the CFBundleExecutable keys, and I was able to upload to App Store Connect. 

TLDR: CFBundleExecutable key removed from the info.plists in the public/node_modules folder, but do not set CFBundlePackageType to BNDL

**Reply by IIxauII** [reply]
I don't understand how modifying a third party installed node module is the answer to this issue? Is this the expected approach to resolve the topic?

**Comment by GeeGYMR** [comment]
I am receiving this error again, but having no luck.

When i search for "CFBundleExecutable" in my xcode it comes up but not in public/node_modules folder instead its in my pod files.

Do i delete this "Executable file" field then try again?.

<img width="962" alt="Screenshot 2024-05-12 at 4 39 49 PM" src="https://github.com/ionic-team/capacitor/assets/136746386/63885d43-482c-425e-80a8-39626d1a3ad7">

**Comment by markope** [comment]
I had the same issue. I fixed it by removing the CFBundleExecutable key and value from the Info.plist of the @capacitor modules in the node_modules folder of my app.

The Info.plist of both the capacitor sub subfolder and the capacitorCordova sub sub folder had to be edited.
